pkg/i18n/i18n.go

```diff
@@ -15,14 +15,22 @@ type Localizer struct {
 }
 
 // NewTranslationSet creates a new Localizer
-func NewTranslationSet(log *logrus.Entry) *TranslationSet {
+func NewTranslationSet(log *logrus.Entry, userConfigLanguage string) *TranslationSet {
 	userLang := detectLanguage(jibber_jabber.DetectLanguage)
 
+	translationSets := GetTranslationSets()
```

Comment:
sorry I missed this on my first review: I think we should be a little more explicit here e.g.:
```go
func NewTranslationSetFromConfig(log *logrus.Entry, configLanguage string) (*TranslationSet, error) {
	if configLanguage == "auto" {
		language := detectLanguage(jibber_jabber.DetectLanguage)
		return NewTranslationSet(log, language), nil
	}
	
	for key := range GetTranslationSets() {
		if key == configLanguage {
			return NewTranslationSet(log, configLanguage), nil
		}
	}

	return nil, errors.New("Language not found: " + configLanguage)
}

func NewTranslationSet(log *logrus.Entry, language string) *TranslationSet {
	log.Info("language: " + userLang)

	baseSet := englishTranslationSet()

	for languageCode, translationSet := range GetTranslationSets() {
		if strings.HasPrefix(userLang, languageCode) {
			_ = mergo.Merge(&baseSet, translationSet, mergo.WithOverride)
		}
	}
	return &baseSet
}
```
What do you reckon?

Reply:
In my opinion, auto is the default option if user configured language isn't matched. This logic is more usable than error reporting.

Reply:
I'm not sure. As a user, if I try to configure the language to be french by setting the config value to 'fr', I would like to get a clear error saying that's not a supported language. By seeing everything render in english, they may think there's a bug with lazygit not registering their language correctly, or they may think the config option is only applied in certain contexts.

Interested to get @mjarkk 's thoughts on this one too

Reply:
In fact, both implementations are fine for me. I mainly want to add a language configuration item. I have updated the code, using an error-reporting implementation.